pages/api.js

```diff
@@ -66,6 +76,70 @@ const API = ({ api, services }) => {
     resume: rate_limiting_resume
   } = rate_limiting || {};
 
+  const handleScroll = throttle(() => {
+    const currentVisibleAnchor = getVisibleAnchor();
+    if (currentVisibleAnchor !== getWindowHash()) {
+      setMenuItem(currentVisibleAnchor);
+      window.history.replaceState(
+        undefined,
+        undefined,
+        `#${currentVisibleAnchor}`
+      );
+    }
+    // approx 8 frames
+  }, 16 * 8);
+
+  const scrollToAnchor = (anchor, smooth = false) => {
+    const anchorElem = document.getElementById(anchor);
+
+    if (!anchorElem) {
+      return;
+    }
+
+    if (smooth) {
+      anchorElem.scrollIntoView({ behavior: "smooth" });
+    } else {
+      anchorElem.scrollIntoView();
+    }
+  };
+
+  const selectAnchor = anchor => {
+    setMenuItem(anchor);
+    scrollToAnchor(anchor);
+  };
+
+  const getVisibleAnchor = () => {
+    if (!contentContainer || !contentContainer.current) {
+      return;
+    }
+
+    const sectionCollection = contentContainer.current.children;
+
+    for (let i = 0; i < sectionCollection.length; i++) {
```

Comment:
I am not found of the for syntax, do you mind using a more functional approach ? for example:

```
const sectionCollection = sectionCollection.find(isElementVisible);
```

Reply:
I didn't see it was an HTMLCollection, so I'd rather use this:

```
const sectionCollection = [...sectionCollection].find(isElementVisible);
```

Your call !

Reply:
I am gonna left it that way as it might be perf sensitive on some terminal and i believe the readibility gain does not worth it in that specific case